packages/perseus-core/src/data-schema.ts

```diff
@@ -1675,6 +1675,10 @@ export type PerseusFreeResponseWidgetScoringCriterion = {
 };
 
 export type PerseusFreeResponseWidgetOptions = {
+    // Whether to allow the user to enter an unlimited number of characters.
+    allowUnlimitedCharacters: boolean;
+    // The maximum number of characters that the user can enter.
+    characterLimit: number;
```

Comment:
Maybe this is a bad idea, but would it make sense to make `characterLimit: number | null`? `null` means "allow unlimited characters" and a number is the the char limit?

Reply:
I had a similar idea originally, the only difference being `0` translated to "unlimited characters" -- there was a discussion with Peter to go with a checkbox for unlimited instead.

Reply:
But in terms of data, it seemed clearer (to me at least) to have this always be a number.